src/main/java/com/ontotext/ehri/deduplication/model/USHMMPersonsFeatureExtractor.java

```diff
@@ -21,196 +13,22 @@
-                statementsMap.get(personId2, "placeBirth"))
-        );
-    }
-
-    private void extractDateBirthFeatures(Alphabet xA, String personId1, String personId2, SparseVector sparseVector) {
-        String dateBirthFirstPerson = statementsMap.get(personId1, "dateBirth");
-        String dateBirthSecondPerson = statementsMap.get(personId2, "dateBirth");
-
-        sparseVector.add(xA.lookupObject("db"), USHMMDate.similarity(dateBirthFirstPerson, dateBirthSecondPerson));
-        sparseVector.add(xA.lookupObject("ldb"), Levenshtein.similarity(dateBirthFirstPerson, dateBirthSecondPerson));
-    }
-
-    private void extractGenderFeature(Alphabet xA, String personId1, String personId2, SparseVector sparseVector) {
-        Set<String> gendersSetFirstPerson = getPersonGenderSet(personId1);
-        Set<String> gendersSetSecondPerson = getPersonGenderSet(personId2);
-        gendersSetFirstPerson.retainAll(gendersSetSecondPerson);
-        sparseVector.add(xA.lookupObject("g"), (gendersSetFirstPerson.size() == 0) ? 0.0d : 1.0d);
-    }
-
-    private Set<String> getPersonGenderSet(String personId) {
-        Set<String> gendersSetPerson = new HashSet<>();
-        gendersSetPerson.add(statementsMap.get(personId, "gender"));
-        gendersSetPerson.add(statementsMap.get(personId, "gender-LinearClass"));
-        gendersSetPerson.add(statementsMap.get(personId, "gender-RuleBased"));
-        return gendersSetPerson;
-    }
-
-    private void extractSourceFeature(Alphabet xA, String personId1, String personId2, SparseVector sparseVector) {
-        String source1 = statementsMap.get(personId1, "sourceId"), source2 = statementsMap.get(personId2, "sourceId");
-        String first = source2, second = source1;
-        if (source1.compareTo(source2) <= 0) {
-            first = source1;
-            second = source2;
-        }
-        sparseVector.add(xA.lookupObject("src1_" + first), 1.0d);
-        sparseVector.add(xA.lookupObject("src2_" + second), 1.0d);
-    }
-
-    private void extractPersonTypeFeature(Alphabet xA, String personId1, String personId2, SparseVector sparseVector) {
-        sparseVector.add(xA.lookupObject("pt"),
-                (statementsMap.get(personId1, "personType").equals(statementsMap.get(personId2, "personType"))) ? 0.0d : 1.0d);
-    }
+    private ClassificationInstance getInstance(USHMMPersonStatementsMapHash statementsMap, USHMMGoldStandardEntry entry, Alphabet xA, Alphabet yA) {
+        USHMMPersonPair personPair = new USHMMPersonPair(entry, statementsMap);
+        USHMMPerson person1 = personPair.getPerson1();
+        USHMMPerson person2 = personPair.getPerson2();
+        USHMMClassificationInstance instance = new USHMMClassificationInstance(xA, person1, person2);
+        return new ClassificationInstance(xA, yA, instance.getSparseVector(), yA.lookupObject(entry.getLabel()));
 
-    private void extractOccupationFeature(Alphabet xA, String personId1, String personId2, SparseVector sparseVector) {
```

Comment:
I didn't notice occupation feature elsewhere, it's good to have it implemented even if it is not used in the current mode.